### PR TITLE
Silk Touch Upgrade for the Destructor

### PR DESCRIPTION
**Pull request by InusualZ**
I was not able to register a place holder texture for the upgrade.

### CODE REVIEW

File: src/main/java/refinedstorage/RSConfig.java [hunk 1]
```diff
@@ -157,6 +158,7 @@ private void loadConfig() {
         craftingUpgradeUsage = config.getInt("crafting", UPGRADES, 5, 0, Integer.MAX_VALUE, "The additional energy used per Crafting Upgrade");
         stackUpgradeUsage = config.getInt("stack", UPGRADES, 12, 0, Integer.MAX_VALUE, "The additional energy used per Stack Upgrade");
         interdimensionalUpgradeUsage = config.getInt("interdimensional", UPGRADES, 1000, 0, Integer.MAX_VALUE, "The additional energy used by the Interdimensional Upgrade");
+        silktouchUpgradeUsage = config.getInt("silktouch", UPGRADES, 15, 0, Integer.MAX_VALUE, "The additional energy used by the Silk Touch Upgrade");
```
Comment (raoulvdberge) [inline src/main/java/refinedstorage/RSConfig.java]: silkTouch

File: src/main/java/refinedstorage/item/ItemUpgrade.java [hunk 2]
```diff
@@ -15,6 +19,7 @@
     public static final int TYPE_CRAFTING = 3;
     public static final int TYPE_STACK = 4;
     public static final int TYPE_INTERDIMENSIONAL = 5;
+    public static final int TYPE_SILKTOUCH = 6;
```
Comment (raoulvdberge) [inline src/main/java/refinedstorage/item/ItemUpgrade.java]: TYPE_SILK_TOUCH

File: src/main/java/refinedstorage/item/ItemUpgrade.java [hunk 3]
```diff
@@ -58,6 +63,8 @@ public static ItemStack getRequirement(int type) {
                 return new ItemStack(Blocks.CRAFTING_TABLE);
             case ItemUpgrade.TYPE_INTERDIMENSIONAL:
                 return new ItemStack(Items.NETHER_STAR);
+            case ItemUpgrade.TYPE_SILK_TOUCH:
+                return Items.ENCHANTED_BOOK.getEnchantedItemStack(new EnchantmentData(Enchantment.getEnchantmentByID(33), 1));
```
Comment (way2muchnoise) [inline src/main/java/refinedstorage/item/ItemUpgrade.java]: You should use `Enchantment.getEnchantmentByLocation("silk_touch")` here as the id might change for what ever reason.

File: src/main/java/refinedstorage/tile/TileDestructor.java [hunk 4]
```diff
@@ -125,23 +125,36 @@ public void updateNode() {
 
                 if (frontStack != null) {
                     if (IFilterable.canTake(itemFilters, mode, compare, frontStack)) {
-                        List<ItemStack> drops = frontBlockState.getBlock().getDrops(worldObj, front, frontBlockState, 0);
+                        if(!upgrades.hasUpgrade(ItemUpgrade.TYPE_SILK_TOUCH)) {
+                            List<ItemStack> drops = frontBlockState.getBlock().getDrops(worldObj, front, frontBlockState, 0);
 
-                        worldObj.playEvent(null, 2001, front, Block.getStateId(frontBlockState));
-                        worldObj.setBlockToAir(front);
+                            for (ItemStack drop : drops) {
```
Comment (way2muchnoise) [inline src/main/java/refinedstorage/tile/TileDestructor.java]: What is this loop for?

**Review by way2muchnoise — CHANGES_REQUESTED** [review]

File: src/main/java/refinedstorage/item/ItemUpgrade.java (hunk 3 above)
Comment (InusualZ) [inline src/main/java/refinedstorage/item/ItemUpgrade.java]: Ohh, thanks :)

File: src/main/resources/assets/refinedstorage/models/item/silktouch_upgrade.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
+  "parent": "item/generated",
+  "textures": {
+    "layer0": "refinedstorage:items/silktouch_upgrade" // Place Holder
```
Comment (raoulvdberge) [inline src/main/resources/assets/refinedstorage/models/item/silktouch_upgrade.json]: silk_touch_upgrade

File: src/main/java/refinedstorage/tile/TileDestructor.java [hunk 6]
```diff
@@ -125,23 +125,36 @@ public void updateNode() {
 
                 if (frontStack != null) {
                     if (IFilterable.canTake(itemFilters, mode, compare, frontStack)) {
-                        List<ItemStack> drops = frontBlockState.getBlock().getDrops(worldObj, front, frontBlockState, 0);
+                        if(!upgrades.hasUpgrade(ItemUpgrade.TYPE_SILK_TOUCH)) {
```
Comment (raoulvdberge) [inline src/main/java/refinedstorage/tile/TileDestructor.java]: is that different if branch really needed? couldn't you just insert have different contents in `drops` depending on the mode?

File: src/main/java/refinedstorage/tile/TileDestructor.java [hunk 7]
```diff
@@ -125,7 +125,20 @@ public void updateNode() {
 
                 if (frontStack != null) {
                     if (IFilterable.canTake(itemFilters, mode, compare, frontStack)) {
-                        List<ItemStack> drops = frontBlockState.getBlock().getDrops(worldObj, front, frontBlockState, 0);
+
+                        List<ItemStack> drops;
+                        if(!upgrades.hasUpgrade(ItemUpgrade.TYPE_SILK_TOUCH)) {
+                            drops = frontBlockState.getBlock().getDrops(worldObj, front, frontBlockState, 0);
+                        } else {
+                            drops = new ArrayList<>(1);
```
Comment (raoulvdberge) [inline src/main/java/refinedstorage/tile/TileDestructor.java]: Use Collections.singletonList here

File: src/main/java/refinedstorage/tile/TileDestructor.java [hunk 8]
```diff
@@ -125,7 +125,20 @@ public void updateNode() {
 
                 if (frontStack != null) {
                     if (IFilterable.canTake(itemFilters, mode, compare, frontStack)) {
-                        List<ItemStack> drops = frontBlockState.getBlock().getDrops(worldObj, front, frontBlockState, 0);
+
```
Comment (raoulvdberge) [inline src/main/java/refinedstorage/tile/TileDestructor.java]: unneeded newline

File: src/main/java/refinedstorage/tile/TileDestructor.java [hunk 9]
```diff
@@ -125,7 +125,20 @@ public void updateNode() {
...
+                        List<ItemStack> drops;
+                        if(!upgrades.hasUpgrade(ItemUpgrade.TYPE_SILK_TOUCH)) {
+                            drops = frontBlockState.getBlock().getDrops(worldObj, front, frontBlockState, 0);
+                        } else {
+                            drops = new ArrayList<>(1);
+                            drops.add(frontStack);
+                        }
+
+                        for (ItemStack drop : drops) {
+                            if(network.insertItem(drop, drop.stackSize, true) != null) {
```
Comment (raoulvdberge) [inline src/main/java/refinedstorage/tile/TileDestructor.java]: space after if

**Pull request merged**